falcon_multiqc/commands/query.py

```diff
@@ -407,7 +414,16 @@ def cli(
         falcon_query = falcon_query.filter(or_(*conditions))
 
     ### ============================== RESULT / OUTPUT =======================================####
-    if falcon_query == None:
+    if len(falcon_query.all()) == 0:
+        for tm in tool_metric:
+            # Check whether tool/metric are correctly spelled.
+            try:
+                if session.query(RawData).filter(RawData.qc_tool == tm[0]).scalar() is None:
+                    raise Exception(f"The tool {tm[0]} is not present in the database, please check its validity.")
+            except MultipleResultsFound:
```

Comment:
Can you explain to me this? When does this get thrown and why

Reply:
If a valid tool is queried (i.e. it's spelled correctly and exists in the database), then the query will return all the rows with that tool, like `select * from rawdata where rawdata.qctool = 'tool'`, and .scalar() only excepts None or 1 result, raises exception for >1 rows returned (maybe there's a rare case where only 1 sample has 1 unique tool, but the cases are low.., I could account for that too)

Reply:
oh wait, if it is 1, it won't trigger the multipleresultfound so it will just continue, so don't worry actually ahha